### PR TITLE
Tweak unnecessary_statements lint to ignore getters; they may have side effects.

### LINKED ISSUE #1015: Update `unnecessary_statements` to report tear-offs but not getters
In #1014, @bwilkerson offered:

> Honestly, I think the right solution is to update unnecessary_statements to stop reporting the use of getters and to report the narrower case of tear-offs. 

and I agree.

### PR DESCRIPTION
**Pull request by davidmorgan**
Fixes #1015

### CODE REVIEW

File: lib/src/rules/unnecessary_statements.dart [hunk 1]
```diff
@@ -160,6 +159,14 @@ class _ReportNoClearEffectVisitor extends UnifyingAstVisitor {
     // Has a clear effect
   }
 
+  @override
+  visitPrefixedIdentifier(PrefixedIdentifier node) {
+    // Allow getters; they may have a side effect.
+    if (node.identifier.bestElement is PropertyAccessorElement) return;
```
[the remark this answers is not in the document]
Comment (davidmorgan) [reply lib/src/rules/unnecessary_statements.dart]: Good catch, thanks. Added test cases for this. (Three, one for each added clause). Fix was to check `isSynthetic`.

I'm not too worried about the cases with overrides; I think they're very rare.

File: lib/src/rules/unnecessary_statements.dart [hunk 2]
```diff
@@ -160,6 +159,14 @@ class _ReportNoClearEffectVisitor extends UnifyingAstVisitor {
     // Has a clear effect
   }
 
+  @override
+  visitPrefixedIdentifier(PrefixedIdentifier node) {
+    // Allow getters; they may have a side effect.
```
[the remark this answers is not in the document]
Comment (davidmorgan) [reply lib/src/rules/unnecessary_statements.dart]: Done, thanks.

**Comment by davidmorgan** [comment]
Thanks!

**Comment by pq** [comment]
Thank you both!

**Pull request merged**